client/client.cpp

```diff
@@ -114,7 +114,7 @@ void Client::socketConnected()
 void Client::resetClientDevice()
 {
     if (m_clientDevice) {
-        m_clientDevice->deleteLater();
+        QMetaObject::invokeMethod(m_clientDevice, "deleteLater", Qt::QueuedConnection);
```

Comment:
Indeed it is 'normally' already queued, though it still make me troubles without invoking it that way, I did not success to find why.
Probably The QtC plugin still making too much call here in a short, but with my latest patches in the plugin, I can not see reentrancy again, though resetClientDevice() is still tried to be called twice in a short (probably due to the different connection pointed to resetClientDevice()).